stdlib/experimental/influxdb/influxdb.go

```diff
@@ -0,0 +1,175 @@
+package influxdb
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	neturl "net/url"
+	"strings"
+	"time"
+
+	"github.com/influxdata/flux"
+	fluxhttp "github.com/influxdata/flux/dependencies/http"
+	fluxurl "github.com/influxdata/flux/dependencies/url"
+	"github.com/influxdata/flux/interpreter"
+	"github.com/influxdata/flux/runtime"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+const (
+	APIFuncName = "api"
+	PackagePath = "experimental/influxdb"
+	APIKind     = PackagePath + "." + APIFuncName
+)
+
+var APISignature = runtime.MustLookupBuiltinType(PackagePath, APIFuncName)
+
+func init() {
+	runtime.RegisterPackageValue(PackagePath, APIFuncName,
+		values.NewFunction(APIFuncName, APISignature, api, true),
+	)
+}
+
+// api submits an HTTP request to the specified API path.
+// Returns HTTP status code, response headers, and body as a byte array.
+func api(ctx context.Context, a values.Object) (values.Value, error) {
+	var (
+		deps      = flux.GetDependencies(ctx)
+		validator fluxurl.Validator
+		client    fluxhttp.Client
+		err       error
+	)
+	{
+		if validator, err = deps.URLValidator(); err != nil {
+			return nil, errors.New("internal error")
+		}
+		if client, err = deps.HTTPClient(); err != nil {
+			return nil, errors.New("internal error")
+		}
+	}
+
+	var (
+		args    = interpreter.NewArguments(a)
+		method  string
+		path    string
+		host    string
+		token   string
+		headers values.Dictionary
+		query   values.Dictionary
+		timeout values.Duration
+		body    []byte
+	)
+	{
+		if method, err = args.GetRequiredString("method"); err != nil {
+			return nil, err
+		}
+
+		if path, err = args.GetRequiredString("path"); err != nil {
+			return nil, err
+		}
+
+		if host, err = args.GetRequiredString("host"); err != nil {
+			return nil, err
+		}
+
+		if token, err = args.GetRequiredString("token"); err != nil {
+			return nil, err
+		}
+
+		if raw, ok := args.Get("timeout"); !ok {
+			timeout = values.ConvertDurationNsecs(30 * time.Second)
+		} else if raw.Type().Nature() != semantic.Duration {
+			return nil, errors.New("timeout argument must be a duration")
+		} else {
+			timeout = raw.Duration()
+		}
+
+		if q, ok := args.Get("query"); ok {
+			query = q.Dict()
```

Comment:
The `Dict` method will panic if `q` is not a Dictionary, can you make an explicit check on the type/nature of `q` before calling `Dict`.